jormungandr-lib/src/interfaces/vote/mod.rs

```diff
@@ -330,7 +330,67 @@ impl TallyResult {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct EncryptedTally(Vec<u8>);
+pub struct EncryptedTally(#[serde(with = "serde_hex_bytes")] Vec<u8>);
+
+mod serde_hex_bytes {
+    use crate::interfaces::vote::EncryptedTally;
+    use serde::de::{Error, Visitor};
+    use serde::{Deserializer, Serialize, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct ByteStringVisitor;
+        impl<'de> Visitor<'de> for ByteStringVisitor {
+            type Value = Vec<u8>;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("Invalid hex encoded encrypted tally")
+            }
```

Comment:
The message will be misleading for binary backends. It should also read like a part of an externally formatted sentence, see [the doc](https://docs.rs/serde/1.0.117/serde/de/trait.Visitor.html#tymethod.expecting)

Reply:
I'd split this trivially into a binary and string visitor implementing only the respective methods, like in my example.

Reply:
Perhaps we should make this serde helper module reused throughout the lib, if we have other byte vector fields that are better represented in hex.

Reply:
Yeah, it would be nice to have a serde sub_module, or sub_crate with this kind of stuff.